WordPress/gutenberg
# Add text-indent supports for paragraph block

## What?

Partially addresses https://github.com/WordPress/gutenberg/issues/37462

This PR introduces the text-indent support and adds it to the paragraph block. This introduces no new UI, we can do that in a follow-up. Themes can make use of the new setting via theme.json.

## Why?


More tools for the theme author's arsenal

## How?


Using the style engine, following how text-transform and text-decoration are being implemented

## Testing Instructions





Add to your theme:

```
			"core/paragraph": {
				"typography": {
					"textIndent": "3em"
				}
			},
```

Insert some paragraphs and check the frontend

## Screenshots or screencast 

Editor:

<img width="1323" alt="Screenshot 2024-03-01 at 11 01 15" src="https://github.com/WordPress/gutenberg/assets/3593343/5200ee35-d3a5-41d7-8845-83eaad14d984">

Frontend:

<img width="1318" alt="Screenshot 2024-03-01 at 11 01 31" src="https://github.com/WordPress/gutenberg/assets/3593343/fd140c15-6148-47ed-845d-e3eff84283ed">



Questions:

- Do we want to have the first paragraph indented? If not, do we allow for an extra control for the users who want all the paragraphs indented?
- What about the margin collapsing? Do we want that, and how would the controls work out?
